test/bridge/limiter/Queue.t.sol

```diff
@@ -0,0 +1,105 @@
+// SPDX-License-Identifier: MIT
+pragma solidity >=0.8.0;
+
+import "forge-std/Test.sol";
+
+import {EnumerableQueueUpgradeable} from "src-bridge/libraries/EnumerableQueueUpgradeable.sol";
+
+contract QueueTest is Test {
+    using EnumerableQueueUpgradeable for EnumerableQueueUpgradeable.KappaQueue;
+
+    EnumerableQueueUpgradeable.KappaQueue internal queue;
+
+    uint256 internal constant AMOUNT = 5;
+
+    function setUp() public {
+        vm.warp(14000000);
+    }
+
+    // Add to Full (F) queue, then poll all
+    function testAdd1F(bytes32 key) public {
+        vm.assume(key != bytes32(0));
+        _add(key, 4 * AMOUNT);
+        _poll_front(4 * AMOUNT);
+    }
+
+    // Add to queue, then poll until it's empty (E) two times
+    function testAdd2E(bytes32 key) public {
+        vm.assume(key != bytes32(0));
+        for (uint256 i = 0; i < 2; ++i) {
+            key = _add(key, 2 * AMOUNT);
+            _poll_front(2 * AMOUNT);
+        }
+    }
+
+    // Add to queue, then poll until it's empty (E) three times
+    function testAdd3E(bytes32 key) public {
+        vm.assume(key != bytes32(0));
+        for (uint256 i = 0; i < 3; ++i) {
+            uint256 amount = i == 0 ? 2 * AMOUNT : AMOUNT;
+            key = _add(key, amount);
```

Comment:
_add -> _addCheck - should be checking the actual state and not just bool return value